spec/controllers/users/edit_password_controller_spec.rb

```diff
@@ -29,33 +29,33 @@
       end
 
       it 're-encrypts PII on active profile' do
-        stub_sign_in
+        user = stub_sign_in
+        user.save! # so uuid and password_salt are generated.
 
-        user = controller.current_user
         profile = create(:profile, :active, user: user, pii: { ssn: '1234' })
 
-        old_password = user.password
         new_password = 'salty new password'
+        old_password = user.password
+        old_hashed_password = user.unlock_password(old_password)
 
         cacher = Pii::Cacher.new(user, controller.user_session)
-        cacher.save(old_password, profile)
+        cacher.save(old_hashed_password, profile)
 
         email_notifier = instance_double(EmailNotifier)
         allow(EmailNotifier).to receive(:new).with(user).and_return(email_notifier)
 
         expect(email_notifier).to receive(:send_password_changed_email)
 
-        params = { password: new_password, current_password: old_password }
+        params = { password: new_password, current_password: user.password }
         patch :update, update_user_password_form: params
 
         profile.reload
+        new_hashed_password = user.unlock_password(new_password)
 
-        expect(profile.decrypt_pii(new_password)).to be_a Pii::Attributes
-        expect(profile.decrypt_pii(new_password).ssn).to eq '1234'
-
-        pending('actual encryption')
+        expect(profile.decrypt_pii(new_hashed_password)).to be_a Pii::Attributes
+        expect(profile.decrypt_pii(new_hashed_password).ssn).to eq '1234'
         expect do
-          profile.decrypt_pii(old_password)
+          profile.decrypt_pii(old_hashed_password)
         end.to raise_error Pii::EncryptionError
```

Comment:
simplified *and* made faster, probably not. To stub in all the encryption pieces is tricky because the controller explicitly saves both the user and the profile object, as well as re-encrypting the PII.

I have simplified (I think) in 45679a0. As far as speed, the run time for this time on my computer is 1.75ms which is pretty quick imo.